README.md

```diff
@@ -36,55 +36,120 @@ Then install the library:
+Install latest Firefox and [Geckodriver](https://github.com/mozilla/geckodriver/releases).
+Make sure to have compatible version of Geckodriver and Firefox!
 
-The `chromedriver` binary must be placed in system `PATH` directory, otherwise you must provide the path when starting Selenium server
-(eg. `java -Dwebdriver.chrome.driver="/path/to/chromedriver" -jar selenium-server-standalone-#.jar`).
+Run `geckodriver` binary (it start to listen on port 4444 by default):
 
-```php
-$driver = RemoteWebDriver::create($host, DesiredCapabilities::chrome());
+```sh
+geckodriver
 ```
 
-##### Launch Firefox
+#### c) Selenium standalone server
 
-Install latest Firefox and [Geckodriver](https://github.com/mozilla/geckodriver/releases).
+[Selenium server](https://selenium.dev/downloads/) is useful especially when you need to execute multiple tests at once
+or your tests are run in different browsers - like on your CI server.
+
+Selenium server receives commands and starts new sessions using browser drivers acting like hub distributing the commands
+amongst multiple nodes.
+
+To run the standalone server, download [`selenium-server-standalone-#.jar` file](http://selenium-release.storage.googleapis.com/index.html)
+(replace # with the current server version). Keep in mind **you must have Java 8+ installed**.
+
+Run the server:
+
+```sh
+java -jar selenium-server-standalone-#.jar
+```
+
+You may need to provide path to `chromedriver`/`geckodriver` binary (if they are not placed in system `PATH` directory):
+
+```sh
+# Chromedriver:
+java -Dwebdriver.chrome.driver="/opt/chromium-browser/chromedriver" -jar vendor/bin/selenium-server-standalone-#.jar
+# Geckodriver:
+java -Dwebdriver.gecko.driver="/home/john/bin/geckodriver" -jar vendor/bin/selenium-server-standalone-#.jar
+
+# (These options could be combined)
+```
 
-The `geckodriver` binary must be placed in system `PATH` directory, otherwise you must provide the path when starting Selenium server
-(eg. `java -Dwebdriver.gecko.driver="/path/to/geckodriver" -jar selenium-server-standalone-#.jar`).
+If you want to distribute browser sessions amongst multiple servers ("grid mode" - one Selenium hub and multiple Selenium nodes) please
+[refer to documentation](https://selenium.dev/documentation/en/grid/).
 
+#### d) Docker
+
+Selenium server could also be started inside Docker container - see [docker-selenium project](https://github.com/SeleniumHQ/docker-selenium).
+
+### 2. Create a Browser Session
+
+When creating a browser session, be sure to pass the url of your running server.
+
+For example:
+
+```php
+// Chromedriver (if started using --port=4444 as above)
+$host = 'http://localhost:4444';
+// Geckodriver
+$host = 'http://localhost:4444';
+// selenium-server-standalone-#.jar (version 2.x or 3.x)
+$host = 'http://localhost:4444/wd/hub';
+// selenium-server-standalone-#.jar (version 4.x)
+$host = 'http://localhost:4444';
+```
+
+Now you can start browser of your choice:
 
 ```php
+// Chrome
```

Comment:
Could we document the use statements so users can copy paste them?

Reply:
Added above